enterprise/reporting/event_stats_test.go

```diff
@@ -0,0 +1,128 @@
+package reporting
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats/memstats"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	mocksBackendConfig "github.com/rudderlabs/rudder-server/mocks/backend-config"
+	"github.com/rudderlabs/rudder-server/utils/pubsub"
+	"github.com/rudderlabs/rudder-server/utils/types"
+)
+
+func TestEventStatsReporter(t *testing.T) {
```

Comment:
Can we add some more tests where we skip for other status like failed or something?

Reply:
The statsstore doesnt have a method to fetch all metrics, So im not sure how to add a test which doesnt get capture. I added a dummy metric to see if it changes values for current metric